compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -1270,9 +1286,12 @@ public void visit(BLangSimpleVarRef varRefExpr) {
                 }
             }
         }
-
         if (!recordFieldDefaultValue && !objectFieldDefaultValueRequiringIsolation && enclInvokable != null &&
                 isReferenceToVarDefinedInSameInvokable(symbol.owner, enclInvokable.symbol)) {
+            if (this.inIsolationStartAction
+                    && !isSubtypeOfReadOnlyOrIsolatedObjectOrInferableObject(symbol.owner, symbol.getType())) {
```

Comment:
Do we have tests where a normal function call is an arg to an isolated start action?

Reply:
added in https://github.com/ballerina-platform/ballerina-lang/pull/41772/commits/3f069ce069ced6817f05c87fb0c77f5f82867da4